### ISSUE TITLE
Album artist grouping broken when not set as the first grouping level

### ISSUE THREAD

**Issue by Clementine-Issue-Importer** [clementine-player/Clementine#2757]
_From [tveloc...@gmail.com](https://code.google.com/u/106914591471623770109/) on February 26, 2012 22:04:17_

What steps will reproduce the problem? 1. Use the "Group by > Advanced grouping..." feature for your library view
2. Select "Album artist" as the first level and "Album" as the second one
3. Compare to: "Genre" as the first level and "Album artist" as the second, "Album" as the third. What is the expected output? What do you see instead? In the last case, I would expect my library to be sorted similarly to the first case (Album artist > Album), only divided under genres. Instead, the Album artist option is broken when not set in the first level, making it behave like Artist.

In other words I would expect my library to be sorted like Genre > Album Artist > Album. Instead I get Genre > Artist > Album. This behavior is particularly broken with large collections of albums. What version of the product are you using? On what operating system? 1.0, 1.0.1, and commit 526a9d99604d2a2e90e7c1f7a26505648af1fce6 on git are all broken. Ubuntu 11.10. Also have used 1.0 on Fedora and Debian. Please provide any additional information below.

_Original issue: http://code.google.com/p/clementine-player/issues/detail?id=2757_
